### PR TITLE
Remove obsolete pipeline code

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- We no longer run our own testing so this
  code is no longer needed.
- Remove PullRequest pipelines as they are no longer in use.
- Combine 'build' and 'pullRequest' variable setup cases
  as they are identical.
- Also remove temp reverse spec map used prior to #2836.

Related #1450
Depends ~#2728~ ~#2638~ ~#2467~ ~#2272~ ~#2836~ ~#4443~
[skip ci]
Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
jenkins test sanity.functional plinux jdk11

**Comment by AdamBrousseau** [comment]
@vsebe for review

File: buildenv/jenkins/common/variables-functions [hunk 1]
```diff
@@ -360,20 +360,6 @@ def set_node(job_type) {
...
-* Set the RELEASE variable with the value provided in the variables file.
-*/
-def set_release() {
-    RELEASE = get_value(VARIABLES."${SPEC}".release, SDK_VERSION)
-}
-
-/*
-* Set the JDK_FOLDER variable with the value provided in the variables file.
-*/
-def set_jdk_folder() {
```
Comment (vsebe) [inline buildenv/jenkins/common/variables-functions]: `set_jdk_folder()` is still required by the build() method - see `stage('Java Version'){...}` in common/build script

File: buildenv/jenkins/common/variables-functions [hunk 2]
```diff
@@ -360,20 +360,6 @@ def set_node(job_type) {
     }
 }
 
-/*
-* Set the RELEASE variable with the value provided in the variables file.
-*/
-def set_release() {
```
Comment (vsebe) [inline buildenv/jenkins/common/variables-functions]: this method is still required - see stage('Java Version'){...} in common/build script

File: buildenv/jenkins/common/variables-functions [hunk 3]
```diff
@@ -385,8 +371,6 @@ def set_build_variables() {
     BOOT_JDK = get_value(VARIABLES."${SPEC}".boot_jdk, SDK_VERSION)
     FREEMARKER = VARIABLES."${SPEC}".freemarker
     OPENJDK_REFERENCE_REPO = VARIABLES."${SPEC}".openjdk_reference_repo
-    set_release()
-    set_jdk_folder()
```
Comment (vsebe) [inline buildenv/jenkins/common/variables-functions]: see comments above (still required - see `stage('Java Version'){...}` in common/build script)

**Comment by AdamBrousseau** [comment]
jenkins compile plinux jdk11

File: buildenv/jenkins/common/variables-functions (hunk 1 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions]: Good catch. Odd I thought I had done a PR build before but it clearly fails now. Going to try `*` in the path and see if that works.

Comment (AdamBrousseau) [reply buildenv/jenkins/common/variables-functions]: That won't work. Added these functions back in.

**Comment by AdamBrousseau** [comment]
jenkins compile plinux jdk11

**Pull request merged**